### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by letrungtrieu**

### CODE REVIEW

File: backend/src/company/company.service.ts [hunk 1]
```diff
@@ -48,12 +48,18 @@ export class CompanyService {
...
+        if (domain)
+            company.domain = domain;
+        if (website)
+            company.website = website;
+        if (address)
+            company.address = address;
+        if (email)
+            company.email = email;
+        if (phone)
+            company.phone = phone;
```
Comment (tanvuongduc) [inline backend/src/company/company.service.ts:62]: this code just do the same as before. U can discard this one

File: backend/src/team/team.controller.ts [hunk 2]
```diff
@@ -5,14 +5,16 @@ import {
     Body,
     Get,
     Query,
+    Param,
 } from '@nestjs/common';
 
 import { TeamService } from './team.service';
 
-@Controller('teams')
+@Controller('team')
```
Comment (tanvuongduc) [inline backend/src/team/team.controller.ts]: s?

File: backend/src/user/user.service.ts [hunk 3]
```diff
@@ -36,6 +38,85 @@ export class UsersService {
         return result.id as String;
     }
 
+    async getUserById(uid: String) {
+        const user = await this.findUserById(uid);
+        return {
+            id: user.id,
```
Comment (tanvuongduc) [inline backend/src/user/user.service.ts:44]: what if user is null-not found?

File: backend/src/user/user.service.ts [hunk 4]
```diff
@@ -36,6 +38,85 @@ export class UsersService {
...
+        phone: String,
+        email: String,
+        password: String,
+        socialNetwork: SocialNetwork,
+        bank: Bank,
+        status: String
+    ) {
+        const updatedUser = await this.findUserById(uid);
+        if (name) {
+            updatedUser.name = name;
```
Comment (tanvuongduc) [inline backend/src/user/user.service.ts:74]: updatedUser.name = name || updatedUser.name

File: backend/src/department/department.controller.ts [hunk 5]
```diff
@@ -57,25 +59,26 @@ export class DepartmentController {
         return res;
     }
 
-    @Get('teams')
+    @Get('teams/:id')
```
Comment (tanvuongduc) [inline backend/src/department/department.controller.ts]: it should be @Get(':id/teams')

**Review by tanvuongduc — CHANGES_REQUESTED** [review]

File: backend/src/user/user.service.ts (hunk 3 above)
Comment (letrungtrieu) [reply backend/src/user/user.service.ts:44]: it throw HttpException in method findUserById

**Review by tanvuongduc — APPROVED** [review]

**Review by tanvuongduc — APPROVED** [review]

**Pull request merged**